components/dashboard/src/projects/ProjectSettings.tsx

```diff
@@ -90,16 +83,73 @@ export default function () {
                 checked={!project.settings?.keepOutdatedPrebuildsRunning}
                 onChange={({ target }) => updateProjectSettings({ keepOutdatedPrebuildsRunning: !target.checked })}
             />
+            <CheckBox
+                title={
+                    <span>
+                        Use Last Successful Prebuild{" "}
+                        <PillLabel type="warn" className="font-semibold mt-2 ml-2 py-0.5 px-2 self-center">
+                            Alpha
+                        </PillLabel>
+                    </span>
+                }
+                desc={
+                    <span>
+                        Skip waiting for prebuilds in progress and use the last successful prebuild from previous
+                        commits on the same branch.
+                    </span>
+                }
+                checked={!!project.settings?.allowUsingPreviousPrebuilds}
+                onChange={({ target }) =>
+                    updateProjectSettings({
+                        allowUsingPreviousPrebuilds: target.checked,
+                        // we are disabling prebuild cancellation when incremental workspaces are enabled
+                        keepOutdatedPrebuildsRunning: target.checked || project?.settings?.keepOutdatedPrebuildsRunning,
```

Comment:
I feel like we'll eventually want to remove this feature. It's not ideal / has too many unfortunate side effects.